### PR TITLE
Extracting out signature request related code from RootRPCMethodsUI conponent

### LINKED ISSUE #5561: Extract Signature Request code from RootRPCMethodsUI
### **Description**

Extract `renderSignModal` function from [RootRPCMethodsUI.js](https://github.com/MetaMask/metamask-mobile/blob/bdb7f37c90e4fc923881a07fca38d4e77c73a579/app/components/Nav/Main/RootRPCMethodsUI.js) into a separate component: 

Message listeners here should also be moved to this new component:

-  https://github.com/MetaMask/metamask-mobile/blob/bdb7f37c90e4fc923881a07fca38d4e77c73a579/app/components/Nav/Main/RootRPCMethodsUI.js#L101 

- https://github.com/MetaMask/metamask-mobile/blob/bdb7f37c90e4fc923881a07fca38d4e77c73a579/app/components/Nav/Main/RootRPCMethodsUI.js#L743

Ensure test coverage for the new code.

### **Acceptance Criteria**
- All of signature request related code is moved to the new component (as much as possible)
- Signature request functionality continues to work as it is currently
- Refactored code has good unit test coverage

### **References**
https://docs.google.com/document/d/19xKY4e07UThOa1jsDQA8pjsNLUBRn39EdUbaFpz3OR0/edit?usp=sharing

### PR DESCRIPTION
**Pull request by jpuri**
Fixes: #5561 

Extract signature request related code from `RootRPCMethodUI.js` into a react component.

QA Steps:
- check to ensure that all signature request pages continue to work as before

### CODE REVIEW

File: app/components/UI/SignatureRequest/Root/Root.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import { MessageInfo, MessageParams, PageMeta } from '../types';
+
+const styles = StyleSheet.create({
+  bottomModal: {
+    justifyContent: 'flex-end',
+    margin: 0,
+  },
+});
+
+// todo: fix type
```
Comment (blackdevelopa) [inline app/components/UI/SignatureRequest/Root/Root.tsx:23]: nit: we should remove this, yeah?

File: app/components/UI/SignatureRequest/Root/Root.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+const Root = () => {
+  const navigation = useNavigation();
+  const { colors } = useTheme();
+
+  const [currentPageMeta, setCurrentPageMeta] = useState<PageMeta>();
+  const [pendingApproval, setPendingApproval] = useState<MessageInfo>();
+  const [showExpandedMessage, setShowExpandedMessage] = useState(false);
```
Comment (blackdevelopa) [inline app/components/UI/SignatureRequest/Root/Root.tsx:30]: nit: we can add type here also

File: app/components/UI/SignatureRequest/Root/Root.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  },
+});
+
+// todo: fix type
+const Root = () => {
+  const navigation = useNavigation();
+  const { colors } = useTheme();
+
+  const [currentPageMeta, setCurrentPageMeta] = useState<PageMeta>();
+  const [pendingApproval, setPendingApproval] = useState<MessageInfo>();
```
Comment (blackdevelopa) [inline app/components/UI/SignatureRequest/Root/Root.tsx:29]: nit: pendingApproval can also be undefined. L40

File: app/components/UI/SignatureRequest/Root/Root.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      type: ApprovalTypes.SIGN_MESSAGE,
+      origin: signMsgParams.origin,
+    });
+  };
+
+  useEffect(() => {
+    Engine.context.MessageManager.hub.on(
+      'unapprovedMessage',
+      (messageParams: MessageParams) => {
+        onUnapprovedMessage(messageParams, 'eth');
```
Comment (blackdevelopa) [inline app/components/UI/SignatureRequest/Root/Root.tsx:61]: nit: can the type `'eth'` be a constant? Same as `personal` and `typed`

File: app/components/UI/SignatureRequest/Root/Root.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+      animationOutTiming={600}
+      onBackdropPress={onSignAction}
+      onBackButtonPress={
+        showExpandedMessage ? toggleExpandedMessage : onSignAction
+      }
+      onSwipeComplete={onSignAction}
+      swipeDirection={'down'}
+      propagateSwipe
+    >
+      {signType === 'personal' && (
```
Comment (blackdevelopa) [inline app/components/UI/SignatureRequest/Root/Root.tsx:108]: nit: can we have signType be a constant

File: app/components/UI/SignatureRequest/types.ts
Comment (blackdevelopa) [inline app/components/UI/SignatureRequest/types.ts:1]: Just confirming, the types here is for all signature types?

File: app/components/Nav/Main/RootRPCMethodsUI.js [hunk 6]
```diff
@@ -820,7 +729,7 @@ const RootRPCMethodsUI = (props) => {
 
   return (
     <React.Fragment>
-      {renderSigningModal()}
+      <SignatureRequestRoot />
```
Comment (blackdevelopa) [inline app/components/Nav/Main/RootRPCMethodsUI.js:732]: 🚀

**Review by blackdevelopa — COMMENTED** [review]
LGTM 🚀 . Just small nitpicking.

File: app/components/UI/SignatureRequest/Root/Root.tsx (hunk 2 above)
Comment (jpuri) [reply app/components/UI/SignatureRequest/Root/Root.tsx:30]: It is inferred as we are initialising the value.

File: app/components/UI/SignatureRequest/Root/Root.tsx (hunk 3 above)
Comment (jpuri) [reply app/components/UI/SignatureRequest/Root/Root.tsx:29]: Yeh that is inferred as initial value is `undefined`

File: app/components/UI/SignatureRequest/types.ts
Comment (jpuri) [reply app/components/UI/SignatureRequest/types.ts:1]: yep

**Review by blackdevelopa — APPROVED** [review]

**Review by segun — APPROVED** [review]

**Comment by seaona** [comment]
@jpuri PR looks good. I can see how the functionality is preserved and signatures are working. 

[mobile-sign.webm](https://user-images.githubusercontent.com/54408225/234544652-feb6c03b-13ec-4490-842b-9116af0fa488.webm)

**Pull request merged**